### ISSUE TITLE
A warning for users running out of browser connections?

### ISSUE THREAD

**Issue by mmzeeman** [zotonic/zotonic#1180]
It would be nice to have a warning for users which open too many tabs to a site and are running out of browser connections because of this.

**Comment by mworrell** [comment]
This is probably due to long polling comet connections?
Or do ws connections also count to the limit?

What we might consider is to switch tabs that are not in focus to *poll with delay*, like nop Ajax postbacks. Actually, aren't we already doing this for the session ping/pong?

If the tab doesn't have focus, we just close the open comet connection and fallback to a periodic ping.

**Comment by mmzeeman** [comment]
Good one… There is an api for this, with its own peculiarities of course, for that. But it works nicely for tabs…

https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API

**Comment by mmzeeman** [comment]
And yes, the websocket of this user was blocked by a corporate firewall.

I don't know about limits to the number websocket connections which can be made.

**Comment by mworrell** [comment]
We need to redraw the connection state machines for this functionality. Maybe also add SSE into it, when we are busy with it anyway...

**Comment by mworrell** [comment]
Preliminary scheduled for 1.0 - maybe merge into 0.13.x when implemented.

**Comment by ddeboer** [comment]
Good stuff! But will fixing it break BC? If not, I prefer to put this on the roadmap milestone instead.

**Comment by mmzeeman** [comment]
Indeed it doesn't break BC, but it is a common stumbling block for end-users. I will probably fix it before 1.0 as it costs our helpdesk time and causes preventable frustration.

**Comment by mworrell** [comment]
It is a good warning. We have seen the same problem with some users at schools.
